qemu/tests/nic_acpi_index.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import random
+import re
+
+from avocado.utils import process
+
+from virttest import utils_net
+from virttest import virt_vm
+from virttest import utils_misc
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    When "acpi-index=N" is enabled, NIC name should always be "ethN"
+
+    1) Boot up guest without nic,remove "biosdevname=0" and "net.ifname=0" from
+    kenrel command line, then reboot guest.
+    2) Hotplug the nic to the same slot as cold plug through monitor cmd
+    3) Check the nic name, the guest nic name enoN == acpi-index=N
+    4) Check if new interface gets ip address
+    5) Ping guest new ip from host
+    6) Repeat steps 2~5 5 times
+    7) Hot plug the nic into a slot diff from cold plug through the monitor cmd
+    8) Repeat steps 2~5 5 times
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+
+    def renew_ip_address(session, mac):
+        ifname = utils_net.get_linux_ifname(session, mac)
+        make_conf = "nmcli connection add type ethernet con-name %s ifname " \
+                    "%s autoconnect yes" % (ifname, ifname)
+        arp_clean = "arp -n|awk '/^[1-9]/{print \"arp -d \" $1}'|sh"
+        session.cmd_output_safe(make_conf)
+        session.cmd_output_safe("ip link set dev %s up" % ifname)
+        session.cmd_output_safe("dhclient -r", timeout=240)
+        session.cmd_output_safe("dhclient %s" % ifname, timeout=240)
+        session.cmd_output_safe(arp_clean)
+
+    def verified_nic_name():
+        ifname = utils_net.get_linux_ifname(session,
+                                            vm.get_mac_address())
+        pattern = int(re.findall(r'\d+', ifname)[-1])
+        nic_name_number = params.get_numeric("nic_name_number")
+        if pattern == nic_name_number:
+            logging.info("nic name match")
+        else:
+            test.fail("nic name doesn't match")
+
+    def ping_test():
+        host_ip = utils_net.get_host_ip_address(params)
+        status, output = utils_net.ping(host_ip, 10, timeout=30)
+        if status:
+            test.fail(
+                "%s ping %s unexpected, output %s" % (vm.name, host_ip, output))
+
+    def get_hotplug_nic_ip(vm, nic, session):
+        def __get_address():
```

Comment:
I think you can use single `_` to define a private method.

Reply:
Changed. Please help me review it again.